### ISSUE TITLE
heroic: GStreamer fails under Heroic with Wine-GE

### ISSUE THREAD

**Issue by aidalgol** [NixOS/nixpkgs#215213]
### Describe the bug
Windows games run under Heroic with [Wine-GE](https://github.com/GloriousEggroll/wine-ge-custom) fail to play video, producing GStreamer errors on the console output.

### Steps To Reproduce
1. Launch any game that plays video cut-scenes when starting a new game (e.g. Blasphemous)
2. Trigger a video cut-scene
3. You should get a black screen instead of video playback.
4. Kill the game
5. Open the game log

### Expected behavior
For in-game videos to play normally.

### Screenshots
Not applicable because the symptom is a black screen.

### Additional context
<details>
<summary>game log</summary>

```console
System Info:
Heroic Version: 2.6.2 Trafalgar Law
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: Version: 0.4

Electron Version: 22.1.0
Chrome Version: 108.0.5359.179
NodeJS Version: 16.17.1

OS: NixOS KERNEL: 6.1.9 ARCH: x64
CPU: AMD Ryzen 9 3950X 16-Core Processor @3.5 GOVERNOR: performance
RAM: Total: 31.3 GiB Available: 6.41 GiB
GRAPHICS: GPU0: NVIDIA GeForce RTX 3080 VRAM: 10015MB DRIVER: 520.56.06 
PROTOCOL: tty

Game Settings: {
	"autoInstallDxvk": true,
	"autoInstallVkd3d": true,
	"preferSystemLibs": false,
	"enableEsync": true,
	"enableFsync": true,
	"nvidiaPrime": false,
	"enviromentOptions": [],
	"wrapperOptions": [],
	"showFps": true,
	"showMangohud": false,
	"useGameMode": false,
	"language": "",
	"wineVersion": {
		"bin": "/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/bin/wine",
		"name": "Wine - Wine-GE-Proton7-34",
		"type": "wine",
		"lib": "/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64",
		"lib32": "/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib",
		"wineserver": "/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/bin/wineserver",
		"wineboot": "/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/bin/wineboot"
	},
	"winePrefix": "/extrapool/aidan/heroic/prefixes/blasphemous",
	"gogSaves": [
		{
			"name": "saves",
			"location": "/extrapool/aidan/heroic/prefixes/blasphemous/dosdevices/c:/users/aidan/AppData/LocalLow/TheGameKitchen/Blasphemous/Savegames"
		}
	]
}

Game launched at: Wed Feb 08 2023 13:28:59 GMT+1300 (New Zealand Daylight Time)

Launch Command: LD_LIBRARY_PATH=/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64:/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib:/run/opengl-driver/lib:/run/opengl-driver-32/lib:/usr/lib:/usr/lib32 LD_PRELOAD= WINEPREFIX=/extrapool/aidan/heroic/prefixes/blasphemous WINEDLLOVERRIDES=winemenubuilder= DXVK_HUD=fps WINEESYNC=1 WINEFSYNC=1 ORIG_LD_LIBRARY_PATH=/run/opengl-driver/lib:/run/opengl-driver-32/lib:/usr/lib:/usr/lib32 GST_PLUGIN_SYSTEM_PATH_1_0=/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0:/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0 WINEDLLPATH=/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/wine:/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/wine /nix/store/m4rh4p5r24nd0afkdzqvq7hvcvkw3fqw-heroic-2.6.2/libexec/heroic/deps/heroic/build/bin/linux/gogdl launch /extrapool/aidan/heroic/Blasphemous 2068474256 --wine /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/bin/wine --platform windows

Game Log:
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
002c:fixme:winediag:LdrInitializeThunk wine-staging 7.0 is a testing version containing experimental patches.
002c:fixme:winediag:LdrInitializeThunk Please mention your exact version when filing bug reports on winehq.org.
00dc:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 00000001812C8B48, 43) stub
00dc:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000140012360, 43) stub
Mono path[0] = 'Z:/extrapool/aidan/heroic/Blasphemous/Blasphemous_Data/Managed'
Mono config path = 'Z:/extrapool/aidan/heroic/Blasphemous/Blasphemous_Data/Mono/etc'
Loading player data from Z:/extrapool/aidan/heroic/Blasphemous/Blasphemous_Data/data.unity3d00dc:fixme:imm:ImmReleaseContext (0000000000030046, 0000000000330AD0): stub
00dc:fixme:wbemprox:client_security_SetBlanket 00000002419C3DA0, 0000000004651050, 10, 0, (null), 3, 3, 0000000000000000, 0x00000000
00dc:fixme:wbemprox:client_security_Release 00000002419C3DA0
00dc:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
00dc:fixme:system:DisplayConfigGetDeviceInfo DISPLAYCONFIG_DEVICE_INFO_GET_TARGET_NAME: stub
00dc:fixme:system:DisplayConfigGetDeviceInfo DISPLAYCONFIG_DEVICE_INFO_GET_TARGET_NAME: stub
0048:fixme:imm:ImeSetActiveContext (0x4803d0, 0): stub
0048:fixme:imm:ImmReleaseContext (0000000000010020, 00000000004803D0): stub
00dc:fixme:wbemprox:client_security_SetBlanket 00000002419C3DA0, 00000000046555D0, 10, 0, (null), 3, 3, 0000000000000000, 0x00000000
00dc:fixme:wbemprox:client_security_Release 00000002419C3DA0
00dc:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
00dc:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
00dc:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
00dc:fixme:wbemprox:client_security_SetBlanket 00000002419C3DA0, 00000000048E6650, 10, 0, (null), 3, 3, 0000000000000000, 0x00000000
00dc:fixme:wbemprox:client_security_Release 00000002419C3DA0
00dc:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
00dc:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
00dc:fixme:bthpropscpl:BluetoothFindFirstDevice (000000000011E5D0 00000000004D1D30): stub!
00dc:fixme:setupapi:CM_Get_Parent 000000000011E4C4 0x00000001 0x00000000 stub
00dc:fixme:setupapi:CM_Get_Parent 000000000011E4C4 0x00000002 0x00000000 stub
02cc:fixme:ver:GetCurrentPackageId (000000003B10FF50 0000000000000000): stub
02dc:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
02dc:fixme:sync:SetWaitableTimerEx (0000000000000790, 000000003BCCF310, 0, 0000000000000000, 0000000000000000, 0000000000000000, 32) semi-stub
0314:fixme:ver:GetCurrentPackageId (000000004130FF50 0000000000000000): stub
0328:fixme:ver:GetCurrentPackageId (00000000453EFF50 0000000000000000): stub
00dc:fixme:bthpropscpl:BluetoothFindFirstDevice (000000000011E610 00000000004D1D30): stub!
00dc:fixme:setupapi:CM_Get_Parent 000000000011E504 0x00000001 0x00000000 stub
00dc:fixme:setupapi:CM_Get_Parent 000000000011E504 0x00000002 0x00000000 stub
02dc:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems

(wine:361): GStreamer-WARNING **: 13:29:35.805: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstdtls.so': libcrypto.so.1.1: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.805: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstsoup.so': libsoup-2.4.so.1: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.805: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstuvch264.so': libgudev-1.0.so.0: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.806: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstlibav.so': libvdpau.so.1: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.806: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstclosedcaption.so': libpangocairo-1.0.so.0: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.806: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstv4l2codecs.so': libgudev-1.0.so.0: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.806: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstspeex.so': libspeex.so.1: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.806: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstflac.so': libFLAC.so.8: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.807: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstttmlsubs.so': libpango-1.0.so.0: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.807: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstvpx.so': libvpx.so.5: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.807: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstopenal.so': libopenal.so.1: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.807: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib64/gstreamer-1.0/libgstwebp.so': libwebp.so.6: cannot open shared object file: No such file or directory

(wine:361): GStreamer-WARNING **: 13:29:35.807: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstflac.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstflac.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.807: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstttmlsubs.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstttmlsubs.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.808: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstv4l2codecs.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstv4l2codecs.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.808: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstuvch264.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstuvch264.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.808: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstwebp.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstwebp.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.808: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstdtls.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstdtls.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.808: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstclosedcaption.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstclosedcaption.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.809: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstvpx.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstvpx.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.809: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstspeex.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstspeex.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.809: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstsoup.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstsoup.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.809: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstopenal.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstopenal.so: wrong ELF class: ELFCLASS32

(wine:361): GStreamer-WARNING **: 13:29:35.809: Failed to load plugin '/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstlibav.so': /home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/lib/gstreamer-1.0/libgstlibav.so: wrong ELF class: ELFCLASS32
winegstreamer error: decodebin0: Your GStreamer installation is missing a plug-in.
winegstreamer error: decodebin0: ../src-gst_base/gst/playback/gstdecodebin2.c(4719): gst_decode_bin_expose (): /GstBin:bin0/GstDecodeBin:decodebin0:
no suitable plugins found:
Missing decoder: H.264 (Constrained Baseline Profile) (video/x-h264, stream-format=(string)avc, alignment=(string)au, level=(string)4.2, profile=(string)constrained-baseline, codec_data=(buffer)0142c02affe1001b6742c02a95b01e0089f97011000003000100000300780da08846e001000468ca8f20, width=(int)1920, height=(int)1080, framerate=(fraction)60/1, pixel-aspect-ratio=(fraction)1/1, chroma-format=(string)4:2:0, bit-depth-luma=(uint)8, bit-depth-chroma=(uint)8, parsed=(boolean)true)

winegstreamer error: decodebin1: Your GStreamer installation is missing a plug-in.
winegstreamer error: decodebin1: ../src-gst_base/gst/playback/gstdecodebin2.c(4719): gst_decode_bin_expose (): /GstBin:bin1/GstDecodeBin:decodebin1:
no suitable plugins found:
Missing decoder: H.264 (Constrained Baseline Profile) (video/x-h264, stream-format=(string)avc, alignment=(string)au, level=(string)4.2, profile=(string)constrained-baseline, codec_data=(buffer)0142c02affe1001b6742c02a95b01e0089f97011000003000100000300780da08846e001000468ca8f20, width=(int)1920, height=(int)1080, framerate=(fraction)60/1, pixel-aspect-ratio=(fraction)1/1, chroma-format=(string)4:2:0, bit-depth-luma=(uint)8, bit-depth-chroma=(uint)8, parsed=(boolean)true)

Launch command: ['/home/aidan/.config/heroic/tools/wine/Wine-GE-Proton7-34/bin/wine', '/extrapool/aidan/heroic/Blasphemous/Blasphemous.exe', '-single-instance']
```
</details>

### Notify maintainers
I am the only maintainer of this package.

### Metadata
```console
❯ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/pf0xc9gryi9dyp4c6zmynnp3blc547z9-nix-info
copying path '/nix/store/pf0xc9gryi9dyp4c6zmynnp3blc547z9-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.9, NixOS, 22.11 (Raccoon), 22.11.20230204.8e82401`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(aidan): `"nixpkgs"`
 - channels(root): `"nixos"`
 - nixpkgs: `/home/aidan/.nix-defexpr/channels/nixpkgs`
```

**Comment by aidalgol** [comment]
@SuperSamus At least for 64-bit games, the problem was missing GStreamer dependencies in the FHS environment.  I don't know of any 32-bit game in my collection suitable for testing that the resolves the issue you described in #207641 with 32-bit games.

**Comment by SuperSamus** [comment]
Note, GStreamer still doesn't work in Heroic if you set the option "Prefer system libraries". Both on 32 bit and 64 bit games.
Setting `GST_PLUGIN_PATH` or similar doesn't fix the problem.
